connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -726,12 +726,14 @@ private void stopTasks(Collection<ConnectorTaskId> ids) {
     private void awaitStopTask(ConnectorTaskId taskId, long timeout) {
         try (LoggingContext loggingContext = LoggingContext.forTask(taskId)) {
             WorkerTask task = tasks.remove(taskId);
-            connectorStatusMetricsGroup.recordTaskRemoved(taskId);
             if (task == null) {
                 log.warn("Ignoring await stop request for non-present task {}", taskId);
                 return;
             }
 
+            if (connectorStatusMetricsGroup != null) {
+                connectorStatusMetricsGroup.recordTaskRemoved(taskId);
+            }
```

Comment:
why not initialize this in the constructor (and only set the herder in the `Worker#start()` via a setter method in `connectorStatusMetricsGroup`)?

Reply:
I was unclear -- the connectorStatusMetricsGroup is not what's Null -- it actually is there. The issue is that the task is cleaned up. This case happens when a startTask() fails, and task is cleaned up in the "catch" block, then restarted and this code is hit again if you restart the failed task. You can see just moving the removal to after the check for `task == null` will be enoughl